src/js/parsers/section.js

```diff
@@ -80,36 +80,27 @@ class SectionParser {
     if (!finished) {
       let childNodes = isTextNode(element) ? [element] : element.childNodes;
 
-      if (this.state.section.isListSection) {
-        this.parseListItems(childNodes);
-      } else {
-        forEach(childNodes, el => {
-          this.parseNode(el);
-        });
-      }
+      forEach(childNodes, el => {
+        this.parseNode(el);
+      });
     }
 
     this._closeCurrentSection();
 
     return this.sections;
   }
 
-  parseListItems(childNodes) {
-    let { state } = this;
-    forEach(childNodes, el => {
-      let parsed = new this.constructor(this.builder).parse(el);
-      let li = parsed[0];
-      if (li && li.isListItem) {
-        state.section.items.append(li);
-      }
-    });
-  }
-
   runPlugins(node) {
     let isNodeFinished = false;
     let env = {
       addSection: (section) => {
-        this._closeCurrentSection();
+        // avoid creating empty paragraphs due to wrapper elements around
+        // parser-plugin-handled elements
+        if (this.state.section.isMarkerable && !this.state.text) {
```

Comment:
This pattern is repeated a couple of times, I wonder if it would make more sense to move it to `_closeCurrentSection` if we're OK with not copying blank markup sections from the source html?

Reply:
> OK with not copying blank markup sections from the source html

To clarify this, the current behaviour with this PR is that blank markup sections are preserved. Eg: `<p>One</p><p></p><p>Two</p>` will result in three markup sections with the middle one being blank.

If we move the duplicated empty-section-removal logic to `_closeCurrentSection` then all blank markup sections would be removed.

Reply:
Scratch that - I was apparently mis-remembering behaviour, empty markup sections are ignored in both the `dom` and `section` parser. I'll add tests for such and resolve this